lib/user_interface.rb

```diff
@@ -84,16 +76,18 @@ def play
   end
 
   def new_score
-    puts message(:new_score).red.bold \
+    puts ' New Score !!'.red.bold \
       if @board.add(@name, @player.tries, @player.time)
+    PressAnyKey.new
+    UserInterface.new
   end
 
   def check_winner
```

Comment:
yep, it detects if we have a winner or a magician (win in 1 trie).
We can let this name or use yours (report_win)